docker/Makefile

```diff
@@ -0,0 +1,86 @@
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Global arguments
+IMG := kubestellar
+NAME := kubestellar
+PLATFORMS := linux/amd64,linux/arm64,linux/ppc64le
+KUBESTELLAR_VERSION := $(shell curl -sL https://github.com/kubestellar/kubestellar/releases/latest | grep "</h1>" | tail -n 1 | sed -e 's/<[^>]*>//g' | xargs)
+
+
+# Build a local container image for the host architecture
+.PHONY: build
+build:
+	$(info Building the KubeStellar container using Dockerfile...)
+	$(info Arguments: IMG=${IMG})
+	$(info Arguments: KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION})
+	# docker buildx build --build-arg="KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION}" --tag ${IMG}:${KUBESTELLAR_VERSION} .
+	docker build --build-arg="KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION}" --tag ${IMG}:${KUBESTELLAR_VERSION} .
+
+
+# Build and push a multi-arch container image (docker buildx is required)
+.PHONY: buildx
+buildx:
+	$(info Building the KubeStellar container using Dockerfile...)
+	$(info Arguments: IMG=${IMG})
+	$(info Arguments: KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION})
+	$(info Arguments: PLATFORMS=${PLATFORMS})
+	- docker buildx create --name kubestellar-builder --use
+	docker buildx build --push --platform ${PLATFORMS} --build-arg="KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION}" --tag ${IMG}:${KUBESTELLAR_VERSION} .
+	- docker buildx rm kubestellar-builder
+
+
+# Run a new container image
+.PHONY: run
+run:
+	$(info Running a new container container...)
+	$(info Arguments: IMG=${IMG})
+	$(info Arguments: KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION})
+	$(info Arguments: NAME=${NAME})
+	mkdir -p ~/.kcp
+	docker run -d \
+		-p 6443:6443 \
+		-v ~/.kcp:/.kcp \
+		--user $(shell id -u) \
+		--name ${NAME} ${IMG}:${KUBESTELLAR_VERSION}
+	docker cp ${IMG}:/kcp-plugins ~/kcp
+	docker cp ${IMG}:/kubestellar ~/kubestellar
+	$(info export KUBECONFIG=${HOME}/.kcp/admin.kubeconfig)
+	$(info export PATH=$$PATH:${HOME}/kcp:${HOME}/kubestellar/bin)
```

Comment:
is `${HOME}/kcp` really needed? as kcp is running in the container.

Reply:
You don't need to mount the kcp folder to run kubestellar. However, you need access to it's KUBECONFIG somehow to do anything with it like `kubectl ws .` At the time I was thinking at the host user use case so I thought that mounting it as a volume was convenient. You can also mount it as a secret or configmap in your cluster.

Once KubeStellar is running, you need to way to intereact with it.
1, syncer yaml need to point to the correct route to access it back
2. who/how do you plan to create objects  in KubeStellar? ... will need the KUBECONFIG access

Reply:
I'm sorry I misunderstood.
I think it depends again on the use case.
The folders `${HOME}/kcp` and `${HOME}/kubestellar/bin` are used to make available to the host user the binaries for manipulating the ws, creating the syncer, etc.  Currently, I included them in the kubesterllar container image so that they can be used from within the container or by the host without the separate need of downloading them.
Perhaps I could be more clear if I understood how you plan to create access kubestellar to create a simple deployment.

For example, if you want to create a "root:abc" workspace you would need to do `kubectl ws roor` and `kuebectl ws create abc`... those two command need access to the KUBECONFIG and the workspace plugin in the kcp folder.

Perhaps @MikeSpreitzer can clarify my fumbling...

Reply:
When a user on the host issues a command like `kubectl ws create imw` this requires kcp's kubectl plugin for `ws` to be on that user's `$PATH`.  The plugin is named `kubectl-ws` and is found in `${HOME}/kcp` (after that `cp` above).